app/mutations/application/createApplicationReviewMutation.ts

```diff
@@ -15,6 +15,18 @@ const mutation = graphql`
         id
         formResultId
         reviewStatus
+        formResultByFormResultId {
+          applicationReviewsByFormResultId {
+            edges {
+              node {
+                ...ApplicationCommentsByForm_applicationReview
+              }
+            }
+          }
+          applicationReview {
+            ...ApplicationReviewContainer_applicationReview
+          }
```

Comment:
Why refetch both the list of all `applicationReviewsByFormResultId` and the current `applicationReview`? Smells like an overfetch

Reply:
applicationReviewsByFormResultId is for refetching details for the comments component
applicationReview is for updating the button component. 

If I remove either, their respective components don't update. 
Solutions?

Reply:
applicationReviewsByFormResultId is for refetching details for the comments component
applicationReview is for updating the button component.
The two components have different render paths.
If I remove either, their respective components don't update.
Solutions?

Reply:
Why not something like...
```diff
  mutation createApplicationReviewMutation(
    $input: CreateApplicationReviewMutationChainInput!
  ) {
    createApplicationReviewMutationChain(input: $input) {
      applicationReview {
-       id
-       formResultId
-       reviewStatus
+       ...ApplicationCommentsByForm_applicationReview
+       ...ApplicationReviewContainer_applicationReview
        formResultByFormResultId {
          applicationReviewsByFormResultId {
            edges {
              node {
-               ...ApplicationCommentsByForm_applicationReview
+               id
              }
            }
          }
          applicationReview {
-           ...ApplicationReviewContainer_applicationReview
+           id
          }
        }
      }
    }
  }
```
that would avoid re-fetching all the fields for all the connections you're trying to update

Reply:
^ still doesn't solve for the fact that `applicationReviewsByFormResultId` isn't using the `@connection` directive, but maybe we don't need it anymore for relay 7?

Reply:
Ensuring the relay store has the expected state is something we need a test for...have @matthieu-foucault or @dleard already written one of those?

Reply:
As far as I understood `@connection` is only for pagination.
There are some mutation tests, but they need to be updated to actually import the mutation defined in the code. Currently, I believe that the mutation code was copy-pasted into the test file.

Reply:
The `@connection` directive annotates any Connection in the store (an array of Edges) such that that specific Connection instance can be retrieved and manipulated using that key

Reply:
This can be helpful if we have the same Connection appearing at multiple depths in the fragment tree; we can dedupe them using the `@connection` directive (or at least we're supposed to be able to.

Reply:
I see. I tried this, but without the nested Id, so it didn't work. This makes sense.